sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/Diagnostics/EventProcessorEventSource.cs

```diff
@@ -0,0 +1,395 @@
+        }
+
+        /// <summary>
+        ///   Indicates the minimum amount of partitions every event processor needs to own when the distribution is balanced.
+        /// </summary>
+        ///
+        /// <param name="count"> Minimum partitions per event processor.</param>
+        ///
+        [Event(7, Level = EventLevel.Informational, Message = "Expected minimum partitions per event processor '{0}'.")]
+        public void MinimumPartitionsPerEventProcessor(int count)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(7, count);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates the list of unclaimed partitions.
+        /// </summary>
+        ///
+        /// <param name="unclaimedPartitions">List of unclaimed partitions.</param>
+        ///
+        [Event(8, Level = EventLevel.Informational, Message = "Unclaimed partitions: '{0}'.")]
+        public virtual void UnclaimedPartitions(string[] unclaimedPartitions)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(8, unclaimedPartitions.ToString());
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an attempt to claim ownership for a specific partition has started.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition whose processing is starting.</param>
+        ///
+        [Event(9, Level = EventLevel.Informational, Message = "Attempting to claim ownership of partition '{0}'.")]
+        public virtual void ClaimOwnership(string partitionId)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(9, partitionId);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that an exception was encountered while claiming ownership for a specific partition.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the Event Hub partition.</param>
+        /// <param name="errorMessage">The message for the exception that occurred.</param>
+        ///
+        [Event(10, Level = EventLevel.Error, Message = "Failed to claim ownership of partition '{0}'. (ErrorMessage: '{1}')")]
+        public virtual void ClaimOwnershipError(string partitionId, string errorMessage)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(10, partitionId, errorMessage ?? string.Empty);
+            }
+        }
+
+        /// <summary>
+        ///   Indicates that the load is unbalanced and this event processor should own more partitions.
+        /// </summary>
+        ///
+        /// <param name="identifier">A unique name used to identify the event processor.</param>
+        ///
+        [Event(11, Level = EventLevel.Informational, Message = "Load is unbalanced and this event processor should own more partitions. (Identifier: '{0}')")]
+        public void ShouldOwnMorePartitions(string identifier)
```

Comment:
Would it be helpful to also log the current ownership count and the calculated minimumOwnedPartitionsCount in this event?

Reply:
I've already added a log for `minimumOwnedPartitionsCount`. I will add one for current ownership count. Thanks :)